salt/modules/zypperpkg.py

```diff
@@ -218,6 +222,22 @@ def stderr(self):
     def pid(self):
         return self.__call_result.get('pid', '')
 
+    def calc_zypper(self):
+        try:
+            zypp_tmp = version('zypper')
+            # zypper version 1.11.34 in SLE12 update supports vendor change for only dist upgrade
+            if version_cmp(zypp_tmp, '1.11.34') >= 0:
+                # zypper version supports vendor change for dist upgrade
+                self.dup_avc = True
+            # zypper version 1.14.8 in SLE15 update supports vendor change in install/patch/upgrading
+            if version_cmp(zypp_tmp, '1.14.8') >= 0:
+                self.inst_avc = True
+            else:
+                log.error("Failed to compare Zypper version")
+        except Exception as ex:
+            log.error("Unable to get Zypper version")
+            log.error(ex)
```

Comment:
This could be unified as:

```suggestion
            log.error("Unable to get Zypper version: {}".format(ex))
```